doc/source/upgrading_guide_upstream.rst

```diff
@@ -0,0 +1,408 @@
+   :depth: 1
+
+.. _percona-server.in-place-upgrade.preliminary.upstream:
+
+Preliminary steps common to all distributions
+--------------------------------------------------------------------------------
+
+.. include:: .res/text/system-service-manager.support.txt
+
+|tip.run-all.root|.
+
+#. Do the full backup (or dump, if possible) a then stop the server. 
+
+   .. code-block:: bash
+
+      $ service mysql stop
+
+The following steps are different for DEB based and RPM based
+distributions. Depending on your operating system, follow the instructions in
+either :ref:`percona-server.mysql.in-place-upgrade.upstream.deb` or in
+:ref:`percona-server.in-place-upgrade.upstream.rpm`.  Then, :ref:`run mysql_upgrade
+<percona-server.mysql-upgrade.run-manually.upstream>`.
+
+.. _percona-server.in-place-upgrade.upstream.deb:
+
+Upgrading from |MySQL| 5.7 for DEB based Linux distributions
+--------------------------------------------------------------------------------
+
+The following steps are relevant to upgrading |MySQL| 5.7 installed on |debian|
+or |ubuntu|.
+
+#. Check which |MySQL| packages are installed:
+
+   .. code-block:: bash
+
+      $ dpkg-query -l 'mysql-*'
+
+#. Remove |MySQL| server leaving data files intact. In the following
+   example, **MYSQL-PACKAGE-NAME** is a placeholder which represents
+   an installed |MySQL| package that you need to remove. For example,
+   if you have |MySQL| Server Community Edition you need to uninstall
+   `mysql-server` at least.
+
+   .. code-block:: bash
+
+      $ apt remove MYSQL-PACKAGE-NAME
+
+#. Install the new server
+
+   .. code-block:: bash
+
+      $ apt install percona-server-server-5.7
+
+   If you intend to use the |TokuDB| storage engine, specify the |TokuDB|
+   package as well:
+
+   .. code-block:: bash
+
+      $ apt install percona-server-server-5.7 percona-server-tokudb-5.7
+
+.. _percona-server.mysql.in-place-upgrade.rpm:
+
+Upgrading from |MySQL| 5.7 for RPM based Linux distributions
+--------------------------------------------------------------------------------
+
+The following steps are relevant to upgrading from |MySQL|
+5.7. |tip.run-all.root|.
+
+
+.. rubric:: Removing packages
+
+#. Check which |MySQL| related packages are installed:
+
+   .. code-block:: bash
+
+      $ rpm -qa | grep '^mysql-'
+
+#. If any |MySQL| packages are found, remove them without dependencies.
+
+   .. code-block:: bash
+
+      $ rpm -qa | grep '^mysql-' | xargs rpm -e --nodeps
+   
+   .. include:: .res/text/important.package.removing-without-dependency.txt
+
+.. rubric:: Installing new packages
+
+#. Install the ``percona-server-server-57`` package
+
+   .. code-block:: bash
+
+      $ yum install Percona-Server-server-57 
+
+   If you intend to use |Percona Server| 5.7 with |TokuDB|, run :program:`yum
+   install` specifying the |TokuDB| package along with
+   ``percona-server-server-57``.
+
+   .. code-block:: bash
+
+      $ yum install percona-server-server-57 percona-server-tokudb-57
+
```

Comment:
rpm package names for 5.6 and 5.7 are upper-case (at least the Percona-Server part)